R/function_return_linter.R

```diff
@@ -4,6 +4,19 @@
 #'   confusing (because assigning to `x` has some side effect that is muddled
 #'   by the dual-purpose expression).
 #'
+#' @examples
+#' # will produce lints
+#' lint(
```

Comment:
suggestion: an example using a `return()` assignment done for side effects

e.g.

https://github.com/r-lib/lintr/blob/854b2c7442008b4d091b72fe52e16995d86d333b/tests/testthat/test-function_return_linter.R#L40

and

https://github.com/r-lib/lintr/blob/854b2c7442008b4d091b72fe52e16995d86d333b/tests/testthat/test-function_return_linter.R#L69-L72

Reply:
I have included these examples. But should I also include versions of these that don't lint? These will be multiline expressions and a bit difficult to show in `text = `.

Reply:
How about like

```
code_lines <- "
e <- new.env()
foo <- function(x) {
  e$val <- x + 1
  return(e$val)
}
"
cat(code_lines)
lint(text = code_lines, function_return_linter())
```

Reply:
Looks good.